### ISSUE TITLE
Error: SNGM not found in `download_osm_data.py`

### ISSUE THREAD

**Issue by energyLS** [pz-max/pypsa-africa#604]
## Checklist

- [x] I am using the current [`main`](https://github.com/pypsa-meets-earth/pypsa-earth/tree/main) branch or the latest [release](https://github.com/pypsa-meets-earth/pypsa-earth/releases). Please indicate.
- [x] I am running on an up-to-date [`pypsa-earth` environment](https://github.com/pypsa-meets-earth/pypsa-earth/blob/main/envs/environment.yaml). Update via `conda env update -f envs/environment.yaml`.

## Describe the Bug

When executing `download_osm_data`, the eo.get_osm_data() fails. Reason: `SNGM` is passed in the list, but it should be a seperated two letter code `SN` and `GM`.

Update 1: See "Error Message with "SN" and "GM" seperated" below: also the seperated countries do not work.
Update 2: According to `earth_osm view regions` it must be "SN-GM". Will continue on this later.

This is also discussed in https://github.com/pypsa-meets-earth/pypsa-earth/issues/40 and partly tackled in https://github.com/pypsa-meets-earth/pypsa-earth/commit/ddf5127a959dc929b684ac2ab8bf8604df979c33 but maybe not implemented in `download_osm_data` ? 

If anyone has some hints how to tackle it, I am happy to do it.

## Error Message with "SNGM"

```bash
ERROR:osm_data_extractor:SNGM not found. check view_region()                                                                                         
Traceback (most recent call last):
  File "/mnt/c/Users/scl38887/Documents/git/pypsa-earth/.snakemake/scripts/tmp3396vnox.download_osm_data.py", line 110, in <module>
    eo.get_osm_data(
  File "/home/energyls/anaconda3/envs/pypsa-earth/lib/python3.10/site-packages/earth_osm/eo.py", line 97, in get_osm_data
    region_tuple_list = [get_region_tuple(rs) for rs in region_list]
  File "/home/energyls/anaconda3/envs/pypsa-earth/lib/python3.10/site-packages/earth_osm/eo.py", line 97, in <listcomp>
    region_tuple_list = [get_region_tuple(rs) for rs in region_list]
  File "/home/energyls/anaconda3/envs/pypsa-earth/lib/python3.10/site-packages/earth_osm/gfk_data.py", line 172, in get_region_tuple
    id = get_id_by_str(region_str)
  File "/home/energyls/anaconda3/envs/pypsa-earth/lib/python3.10/site-packages/earth_osm/gfk_data.py", line 161, in get_id_by_str
    raise KeyError(f'{region_str} not found. check view_region()')
KeyError: 'SNGM not found. check view_region()'
```

## Error Message with "SN" and "GM" seperated
```
ERROR:osm_data_extractor:SN not found. check view_region()                                                                                           
Traceback (most recent call last):
  File "/mnt/c/Users/scl38887/Documents/git/pypsa-earth/.snakemake/scripts/tmpn8vbj87j.download_osm_data.py", line 119, in <module>
    eo.get_osm_data(
  File "/home/energyls/anaconda3/envs/pypsa-earth/lib/python3.10/site-packages/earth_osm/eo.py", line 97, in get_osm_data
    region_tuple_list = [get_region_tuple(rs) for rs in region_list]
  File "/home/energyls/anaconda3/envs/pypsa-earth/lib/python3.10/site-packages/earth_osm/eo.py", line 97, in <listcomp>
    region_tuple_list = [get_region_tuple(rs) for rs in region_list]
  File "/home/energyls/anaconda3/envs/pypsa-earth/lib/python3.10/site-packages/earth_osm/gfk_data.py", line 172, in get_region_tuple
    id = get_id_by_str(region_str)
  File "/home/energyls/anaconda3/envs/pypsa-earth/lib/python3.10/site-packages/earth_osm/gfk_data.py", line 161, in get_id_by_str
    raise KeyError(f'{region_str} not found. check view_region()')
KeyError: 'SN not found. check view_region()'
```

**Comment by pz-max** [comment]
Fix is coming with https://github.com/pypsa-meets-earth/pypsa-earth/pull/579

**Comment by davide-f** [comment]
Closed via #579
